osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcess.java

```diff
@@ -110,6 +137,8 @@ public void run() {
           Arrays.asList(E_METER_ENCRYPTION, E_METER_AUTHENTICATION));
     } catch (final Exception e) {
       throw new RecoverKeyException(e);
+    } finally {
+      this.deviceKeyProcessingService.stopProcessing(this.deviceIdentification);
```

Comment:
It looks like a potential issue that `startProcessing` is part of a different try-block then `stopProcessing`.
For instance in case of a throttling exception during `canConnectUsingNewKeys` code in the finally-block where `stopProcessing` is located will not be executed. There appears to be no reason to hold the "key operation lock" in such a scenario.

Reply:
documented

Reply:
It looks like there is a typo in the documentation comment ("trhe").
Because there are a number of sentences in the comment, it would be cleaner to start with a capital letter.